packages/core/chrome/core-chrome-browser-internal/src/ui/project/header.tsx

```diff
@@ -0,0 +1,90 @@
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { Router } from 'react-router-dom';
+import { EuiHeader, EuiHeaderLogo, EuiHeaderSection, EuiHeaderSectionItem } from '@elastic/eui';
+import {
+  ChromeBreadcrumb,
+  ChromeGlobalHelpExtensionMenuLink,
+  ChromeHelpExtension,
+  ChromeNavControl,
+} from '@kbn/core-chrome-browser/src';
+import { Observable } from 'rxjs';
+import { MountPoint } from '@kbn/core-mount-utils-browser';
+import { InternalApplicationStart } from '@kbn/core-application-browser-internal';
+import { HeaderBreadcrumbs } from '../header/header_breadcrumbs';
+import { HeaderActionMenu } from '../header/header_action_menu';
+import { HeaderHelpMenu } from '../header/header_help_menu';
+import { HeaderNavControls } from '../header/header_nav_controls';
+import { ProjectNavigation } from './navigation';
+
+interface Props {
+  breadcrumbs$: Observable<ChromeBreadcrumb[]>;
+  actionMenu$: Observable<MountPoint | undefined>;
+  kibanaDocLink: string;
+  globalHelpExtensionMenuLinks$: Observable<ChromeGlobalHelpExtensionMenuLink[]>;
+  helpExtension$: Observable<ChromeHelpExtension | undefined>;
+  helpSupportUrl$: Observable<string>;
+  kibanaVersion: string;
+  application: InternalApplicationStart;
+  navigation: JSX.Element;
+  navControlsRight$: Observable<ChromeNavControl[]>;
+}
+
+export const ProjectHeader = ({
+  application,
+  kibanaDocLink,
+  kibanaVersion,
+  navigation,
+  ...observables
+}: Props) => {
+  const renderLogo = () => (
+    <EuiHeaderLogo
+      iconType="logoElastic"
+      href="#"
+      onClick={(e) => e.preventDefault()}
+      aria-label="Go to home page"
+    />
+  );
+
+  return (
+    <>
+      <EuiHeader position="fixed">
+        <EuiHeaderSection grow={false}>
+          <EuiHeaderSectionItem border="right">{renderLogo()}</EuiHeaderSectionItem>
+          <EuiHeaderSectionItem>
+            <HeaderBreadcrumbs breadcrumbs$={observables.breadcrumbs$} />
+          </EuiHeaderSectionItem>
+        </EuiHeaderSection>
+        <EuiHeaderSection side="right">
+          <EuiHeaderSectionItem>
+            <HeaderHelpMenu
+              globalHelpExtensionMenuLinks$={observables.globalHelpExtensionMenuLinks$}
+              helpExtension$={observables.helpExtension$}
+              helpSupportUrl$={observables.helpSupportUrl$}
+              kibanaDocLink={kibanaDocLink}
+              kibanaVersion={kibanaVersion}
+              navigateToUrl={application.navigateToUrl}
+            />
+          </EuiHeaderSectionItem>
+          <EuiHeaderSectionItem border="left">
+            <HeaderActionMenu actionMenu$={observables.actionMenu$} />
+          </EuiHeaderSectionItem>
+
+          <EuiHeaderSectionItem>
+            <HeaderNavControls navControls$={observables.navControlsRight$} />
+          </EuiHeaderSectionItem>
+        </EuiHeaderSection>
+      </EuiHeader>
+      <Router history={application.history}>
```

Comment:
❤️

Reply:
This likely needs to change to core router in prod.  @vadimkibana and @lukeelmers , can you weigh in?

Reply:
Not completely sure, would need to play with the code, but it might be already good. As the `application.history` is the one from "core", so it sort of already is the "core router".

Reply:
I believe the `history` here is pulling from core's internal application service contract, which means it is the global (unscoped) history instance that's currently only accessible to core.

We'll need to think about what we want to expose here for the project navs in serverless. Since the navs could be anything, it may be hard to avoid leaking the global history. cc @pgayvallet @rudolf

Reply:
The project nav isn't / shouldn't be scoped to a given plugin path, so using the global history seems fine to me, especially given its usage will be isolated in Core code / packages